__tests__/redux/actions/bookmark.test.js

```diff
@@ -0,0 +1,86 @@
+/* eslint-disable no-unreachable */
+/* eslint-disable import/no-named-as-default */
+/* eslint-disable no-undef */
+import configureStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import moxios from 'moxios';
+import { getBoomarks, bookmark } from '../../../src/redux/actions/article.actions';
+import { bookmarkk } from '../../../__mocks__/data';
+import getArticleReducer from '../../../src/redux/reducers/articles.reducer';
+import { GET_BOOKMARK, BOOKMARK } from '../../../src/redux/actions/types/article.type';
+
+const middleware = [thunk];
+const mockStore = configureStore(middleware);
+const store = mockStore({});
+
+describe('SHOULD TEST BOOKMARKS', () => {
+  beforeEach(() => {
+    moxios.install();
+  });
+  afterEach(() => {
+    moxios.uninstall();
+    store.clearActions();
+  });
+  test('should get bookmarks', async () => {
+    moxios.wait(() => {
+      const request = moxios.requests.mostRecent();
+      request.respondWith({
+        status: 200,
+        response: {
+          data: { Bookmark: [], count: 1 },
+        },
+      });
+    });
+    return store.dispatch(getBoomarks(1)).then(() => {
+      expect(store.getActions().length).toEqual(1);
+    });
+  });
+  test('should create bookmark', async () => {
+    moxios.wait(() => {
+      const request = moxios.requests.mostRecent();
+      request.respondWith({
+        status: 200,
+        response: {
+          data: { bookmark: {}, count: 1 },
+        },
+      });
+    });
+    return store.dispatch(bookmark(1)).then(() => {
+      expect(store.getActions().length).toEqual(1);
+    });
+  });
+  test('should not create bookmark', async () => {
+    moxios.wait(() => {
+      const request = moxios.requests.mostRecent();
+      request.respondWith({
+        status: 404,
+        response: {
+          data: { bookmark: 'error' },
+        },
+      });
+    });
+    return store.dispatch(bookmark(1)).then(() => {
+      expect(store.getActions().length).toEqual(1);
+    });
+  });
+});
+describe('Reducer test bookmarking', () => {
```

Comment:
@EmyRukundo wouldn't be better to separate actionsTests & reducerTests into different testsFolder/testFiles?

Reply:
I am going to work on it